SentinelHub/tests/test_widget.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+pytest.importorskip("qgis.core")
+
+from qgis.core import QgsApplication  # noqa: E402
+from qgis.PyQt.QtCore import QPoint, Qt  # noqa: E402
+from qgis.PyQt.QtTest import QTest  # noqa: E402
+from qgis.PyQt.QtWidgets import QAbstractButton  # noqa: E402
+
+from ..dockwidget import SentinelHubDockWidget  # noqa: E402
+
+
+def click_button(button: QAbstractButton) -> None:
+    QTest.mouseClick(
+        button,
+        Qt.LeftButton,
+        pos=QPoint(0, int(button.height() / 2)),
+    )
+
+
+def test_widget_creation(qgis_app: QgsApplication) -> None:
+    sh_widget = SentinelHubDockWidget()
+    assert sh_widget is not None
+
+    sh_widget.show()
+    assert sh_widget.isVisible()
+
+    sh_widget.close()
+    qgis_app.processEvents()
+    assert not sh_widget.isVisible()
+
+
+def test_widget_button() -> None:
+    sh_widget = SentinelHubDockWidget()
+    click_button(sh_widget.currentExtentRadioButton)
+
+    assert sh_widget.currentExtentRadioButton.isChecked()
+    assert not sh_widget.customExtentRadioButton.isChecked()
+
+    click_button(sh_widget.customExtentRadioButton)
+
+    assert not sh_widget.currentExtentRadioButton.isChecked()
+    assert sh_widget.customExtentRadioButton.isChecked()
+
+    sh_widget.close()
```

Comment:
how important is this? it might not happen if the test fails, consider a yielding fixture like you did with `qgis_app`

Reply:
yes was my first option, you are right then I considered also yielding the app and the widget in the same fixture as it was some conflict with the Qt Threads. But I solved and scoped the widget fixture to the function level.